Libraries/LibGfx/ImageFormats/ICOLoader.cpp

```diff
@@ -68,22 +70,40 @@ struct ICOLoadingContext {
     State state { NotDecoded };
     u8 const* data { nullptr };
     size_t data_size { 0 };
+    u16 file_type { 0 };
     Vector<ICOImageDescriptor> images;
     size_t largest_index;
 };
 
-static ErrorOr<size_t> decode_ico_header(Stream& stream)
+static ErrorOr<size_t> decode_ico_header(Stream& stream, u16& out_type)
 {
     auto header = TRY(stream.read_value<ICONDIR>());
-    if (header.must_be_0 != 0 || header.must_be_1 != 1)
-        return Error::from_string_literal("Invalid ICO header");
+    if (header.must_be_0 != 0 || (header.type != 1 && header.type != 2))
```

Comment:
Instead of using the literals `1` and `2` here, could you add an `enum class IconType : u16 { .. }` for these and use that in ICONDIR and `decode_ico_direntry` as well?